server/src/main/java/org/opensearch/tasks/consumer/TopNSearchTaskTracker.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.tasks.consumer;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.search.SearchShardTask;
+import org.opensearch.common.collect.Tuple;
+
+import java.time.Instant;
+import java.util.Comparator;
+import java.util.PriorityQueue;
+import java.util.Queue;
+
+// TODO: Rename this class and add java docs.
+// TODO: Do we want to introduce locks? Multiple threads reporting completed tasks can cause state issues.
+public class TopNSearchTaskTracker {
+    private static final Logger LOGGER = LogManager.getLogger("task.detailslog");
+    private static final TopNSearchTaskTracker INSTANCE = new TopNSearchTaskTracker();
+    // Logs top 5 search request with the highest memory consumption in the last 60 sec.
+    private static final int TOP_QUERIES_SIZE = 5;
+    private static final long LOGGING_FREQUENCY_IN_SECONDS = 60;
+
+    // Can make this blocking - but accuracy can be trade off for speed.
+    private final Queue<Tuple<Long, SearchShardTask>> topQueries = new PriorityQueue<>(
+        TOP_QUERIES_SIZE,
+        Comparator.comparingLong(Tuple::v1)
+    );
+
+    private long lastReportedTime;
+
+    /**
+     * This method returns static instance of this class
+     *
+     * @return singleton instance for class
+     */
+    public static TopNSearchTaskTracker getInstance() {
+        return INSTANCE;
+    }
+
+    private TopNSearchTaskTracker() {
+        this.lastReportedTime = Instant.now().toEpochMilli();
```

Comment:
lastReportedTime initialized to a value when class is loaded, is this correct?

Reply:
Right. Modified it.